core/src/main/java/software/amazon/awssdk/core/sync/ResponseBytes.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.sync;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.IoUtils;
+
+/**
+ * An in-memory representation of the service's response from a streaming operation. This usually obtained by calling the "bytes"
+ * form of a streaming operation, like S3's {@code getObjectBytes}. Can also be retrieved by passing
+ * {@link StreamingResponseHandler#toBytes()} to a streaming output operation.
+ */
+@SdkPublicApi
+public final class ResponseBytes<ResponseT> {
+    private final ResponseT response;
+    private final byte[] responseBytes;
+
+    public ResponseBytes(ResponseT response, InputStream responseStream) throws IOException {
+        this.response = response;
+        this.responseBytes = IoUtils.toByteArray(responseStream);
+    }
+
+    /**
+     * @return the unmarshalled response object from the service.
+     */
+    public ResponseT response() {
+        return response;
+    }
+
+    /**
+     * @return The output from the streaming operation as a byte buffer.
+     */
+    public ByteBuffer asByteBuffer() {
+        return ByteBuffer.wrap(responseBytes).asReadOnlyBuffer();
+    }
+
+    /**
+     * Retrieve the output of the streaming operation as a string.
+     *
+     * @param charset The charset of the string.
+     * @return The output from the streaming operation as a string.
+     */
+    public String asString(Charset charset) {
+        return new String(responseBytes, charset);
+    }
```

Comment:
should we have another `asString()` which uses a default Charset `UTF-8`?

Reply:
We could have an `asUtf8String()`, but that's pretty much the same as `asString(UTF_8)` (with static imports). I'd personally expect `asString()` to use the default character encoding.

Reply:
+1 for toUtf8String, that's consistent with the async stuff

https://github.com/aws/aws-sdk-java-v2/blob/master/core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseHandler.java#L123

Reply:
Two votes beats my one 👍 . Will add!